Kernel/FileSystem/SysFS/Subsystems/DeviceIdentifiers/DeviceComponent.cpp

```diff
@@ -10,11 +10,11 @@
 
 namespace Kernel {
 
-NonnullLockRefPtr<SysFSDeviceComponent> SysFSDeviceComponent::must_create(Device const& device)
+NonnullRefPtr<SysFSDeviceComponent> SysFSDeviceComponent::must_create(Device const& device)
 {
     // FIXME: Handle allocation failure gracefully
     auto device_name = MUST(KString::formatted("{}:{}", device.major(), device.minor()));
-    return adopt_lock_ref_if_nonnull(new SysFSDeviceComponent(move(device_name), device)).release_nonnull();
+    return adopt_ref_if_nonnull(new SysFSDeviceComponent(move(device_name), device)).release_nonnull();
```

Comment:
slightly out of scope:
`_if_nonnull` suggests that you do the check here; maybe `adopt_nonnull_ref` would be better here instead? 🤔 (if i get it right, of course!)

Reply:
This is not the actual functionality for this statement - it says that we adopt a reference if the given pointer is not null (`new` with `nothrow` will return a null pointer on allocation failure), otherwise there's no reference. the result of both of these flows is a given RefPtr.

Reply:
We should probably have a discussion about this in the Discord server as this is not related to this PR.

Reply:
thatʼs not really necessary!—consider that resolved. 🙇🏻‍♂️